Force -O3 for executorch op_div.cpp in "clang 17" also

akrieger opened the pull request:
Summary:
Some platforms use clang 19 under the hood but are setting the clang 17
constraint for bad reasons.

Differential Revision: D80980948

swolchok reviewed (approved):
as long as it passes OSS CI it seems fine

akrieger committed:
Force -O3 for executorch op_div.cpp in "clang 17" also (#13660)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/13660

Some platforms use clang 19 under the hood but are setting the clang 17
constraint for bad reasons.

Reviewed By: swolchok

Differential Revision: D80980948
